dhnx/optimization.py

```diff
@@ -33,35 +42,522 @@ def get_results(self):
 
 
 class OemofInvestOptimizationModel(InvestOptimizationModel):
-    r"""
+    """
     Implementation of an invest optimization model using oemof-solph.
+
+    ...
+
+    Attributes
+    ----------
+    settings : dict
+        Dictionary holding the optimisation settings. See .
+    invest_options : dict
+        Dictionary holding the investment options for the district heating system.
+    nodes : list
+        Empty list for collecting all oemof.solph nodes.
+    buses : dict
+        Empty dictionary for collecting all oemof.solph.Buses of the energy system.
+    es : oemof.solph.EnergySystem
+        Empty oemof.solph.EnergySystem.
+    om : oemof.solph.Model
+        Attribute, which will be the oemof.solph.Model for optimisation.
+    oemof_flow_attr : set
+        Possible flow attributes, which can be used additionally:
+        {'nominal_value', 'min', 'max', 'variable_costs', 'fix'}
+    results : dict
+        Empty dictionary for the results.
+
+    Methods
+    -------
+    check_input():
+        Performs checks on the input data.
+    complete_exist_data():
+        Sets the investment status for the results dataframe of the edges.
+    get_pipe_data():
+        Adds heat loss and investment costs to edges dataframe.
+    setup_oemof_es():
+        The energy system *es* is build.
+    setup():
+        Calls *check_input()*, *complete_exist_data()*, *get_pipe_data()*, and *setup_oemof_es()*.
+
     """
-    def __init__(self, thermal_network):
+    def __init__(self, thermal_network, settings, investment_options):
+
+        self.settings = settings
+        self.invest_options = investment_options
+        self.nodes = []  # list of all nodes
+        self.buses = {}  # dict of all buses
+        self.es = solph.EnergySystem()
+        self.om = None
+
+        # list of possible oemof flow attributes, e.g. for producers source
+        self.oemof_flow_attr = {'nominal_value', 'min', 'max',
+                                'variable_costs', 'fix'}
+
         super().__init__(thermal_network)
         self.results = {}
 
+    def check_input(self):
+        """Check 1:
+
+        Firstly, it is checked, if there are any not-allowed connection in the *edge* data.
+        The following connections are not allowed:
+
+          * consumer -> consumer
+          * producer -> producer
+          * producer -> consumer
+          * consumer -> fork
+
+        An error is raised if one of these connection occurs.
+
+        Check 2:
+
+        Check, if the columns of the heat demand series (if given) have the datatype *int*.
```

Comment:
Why is float not allowed?

Reply:
now, ids of consumers, producers, etc. do not need to be of type *int* anymore